### PR TITLE
feat: non-null unique short name columns [DHIS2-14526]

### PR DESCRIPTION
**Pull request by jbee**
### Summary
Makes `shortName` a mandatory and unique property for the following object types:

* `ProgramTrackedEntityAttributeGroup`
* `DataSet`
* `OrganisationUnitGroup`
* `CategoryOption`
* `Constant`
* `DataElementGroup`

As a consequence any test setup creating or updating these objects needed to provide a `shortName`.

### Automatic Testing
Existing tests were adjusted.

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/category/CategoryOption.java [hunk 1]
```diff
@@ -91,6 +91,7 @@ public CategoryOption()
     public CategoryOption( String name )
     {
         this.name = name;
+        this.shortName = name;
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/category/CategoryOption.java:94]: I assume this is just for testing purposes @jbee ? This is a little bit risky perhaps as we might see unintended consequences. If it is for test purposes we might want to update `DhisConvenienceTest` methods. We need to coordinate with front-end/maintenance app I assume before merging this PR.

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v40/V2_40_16__Unique_non_null_short_names.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        throws Exception
+    {
+        copyUniqueValue( context, "program_attribute_group", "programtrackedentityattributegroupid", "name",
+            "shortname", 50 );
+        copyUniqueValue( context, "dataset", "datasetid", "name", "shortname", 50 );
+        copyUniqueValue( context, "orgunitgroup", "orgunitgroupid", "name", "shortname", 50 );
+        copyUniqueValue( context, "dataelementcategoryoption", "categoryoptionid", "name", "shortname", 50 );
+        copyUniqueValue( context, "constant", "constantid", "name", "shortname", 50 );
+        copyUniqueValue( context, "dataelementgroup", "dataelementgroupid", "name", "shortname", 50 );
+    }
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v40/V2_40_16__Unique_non_null_short_names.java:62]: Looks good!

**Review by larshelge — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/category/CategoryOption.java (hunk 1 above)
Comment (jbee) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/category/CategoryOption.java:94]: It is testing only but I can also update on the call side instead.

Comment (jbee) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/category/CategoryOption.java:94]: And the `default` option uses this. And lots of callers in tests.

**Review by netroms — APPROVED** [review]

**Pull request merged**